src/v/coproc/router.cc

```diff
@@ -112,19 +125,32 @@ ss::future<> router::do_route() {
           }
           return acc;
       };
-    return ss::map_reduce(
-             _sources.begin(),
-             _sources.end(),
-             [this](
-               std::pair<const model::ntp, ss::lw_shared_ptr<topic_state>>& p) {
-                 return route_ntp(p.first, p.second);
-             },
-             std::vector<process_batch_request::data>(),
-             std::move(reducer))
-      .then([this](std::vector<process_batch_request::data> batch) {
-          return process_batch(std::move(batch));
-      })
-      .then([this, next_loop = _jitter()] { _loop_timer.rearm(next_loop); });
+    auto mapper =
```

Comment:
why make a function instead of a free func tho. seems like this can be out of line.